File: BM.BLL/ReaderCategoryManager.cs
using System.Data;
using BookManagement.Model;
using BookManagement.DAL;

namespace BookManagement.BLL
{
    public class ReaderCategoryManager
    {
        private ReaderCategoryService categoryService = new ReaderCategoryService();

        public DataTable GetAllCategories()
        {
            return categoryService.GetAllCategories();
        }

        public bool AddCategory(ReaderCategory category)
        {
            // 业务规则验证
            if (string.IsNullOrEmpty(category.CategoryName))
                throw new System.Exception("类别名称不能为空！");
            if (category.MaxBorrowCount <= 0)
                throw new System.Exception("最大借阅数量必须大于0！");
            if (category.BorrowDays <= 0)
                throw new System.Exception("借阅天数必须大于0！");
            if (category.LateFeePerDay < 0)
                throw new System.Exception("逾期费用不能为负数！");

            return categoryService.AddCategory(category) > 0;
        }

        public bool UpdateCategory(ReaderCategory category)
        {
            // 同样验证
            if (string.IsNullOrEmpty(category.CategoryName))
                throw new System.Exception("类别名称不能为空！");
            // ... 其他验证
            return categoryService.UpdateCategory(category) > 0;
        }

        public bool DeleteCategory(int categoryId)
        {
            return categoryService.DeleteCategory(categoryId) > 0;
        }
    }
}

File: Python/bll/reader_category_manager.py
from dal.reader_category_service import ReaderCategoryService
from models.reader_category import ReaderCategory


class ReaderCategoryManager:
    def __init__(self):
        self.category_service = ReaderCategoryService()

    def get_all_categories(self):
        return self.category_service.get_all_categories()

    def add_category(self, category: ReaderCategory) -> bool:
        if not category.category_name:
            raise Exception("类别名称不能为空！")
        if category.max_borrow_count <= 0:
            raise Exception("最大借阅数量必须大于0！")
        if category.borrow_days <= 0:
            raise Exception("借阅天数必须大于0！")
        if category.late_fee_per_day < 0:
            raise Exception("逾期费用不能为负数！")
        return self.category_service.add_category(category) > 0

    def update_category(self, category: ReaderCategory) -> bool:
        if not category.category_name:
            raise Exception("类别名称不能为空！")
        return self.category_service.update_category(category) > 0

    def delete_category(self, category_id: int) -> bool:
        return self.category_service.delete_category(category_id) > 0